new responsive site

edswf1 opened the pull request:
this site is a fully functional, responsive version of the current dogecoin.com site base on Bootstrap and jQuery.
live version of the site: https://edswf1.github.io/dogecoin.com/
i18n is not included yet, however all other functionality, including tutorial, is.

All images, js and css resources where moved to an assets folder, dogechain api is used to display pricing.

This site has been tested for perfomance, see mobile score below.
<img width="746" alt="perfomance" src="https://user-images.githubusercontent.com/80992067/111922724-a2c02f00-8a9b-11eb-9f95-50bf3f07a94a.png">

Libraries used:
- Bootstrap 4.5.3
- jQuery 3.5.1
- jQuery Easing 1.4.1
- FontAwesome 5.15.0
- particles.js
- Google Fonts Montserrat & Lato

i did not remove any project files, only index.html would be replaced

patricklodder commented:
Hi! Thanks! How does this integrate with the internationalization?

edswf1 commented:
> Hi! Thanks! How does this integrate with the internationalization?

unfortunately it doesn‘t right now. i‘m planning to integrate i18n tomorrow

edswf1 commented:
@patricklodder 
added internationalization with jQuery.i18n.
Language is guessed by browser language (english fallback) and can be set in footer.
Once Language is set, current Language ist stored in sessionStorage (for maintaining language when browsing to guide.html)
added some texts to en,de,fr and es.json, however im not fluent in fr and es.

site is live at: https://edswf1.github.io/dogecoin.com-1/

patricklodder commented:
Thank you! To keep momentum going, I'm going to try to give you feedback in iterations. This is going to make me look like a terrible nitpicker that is never satisfied 😂 - but the review is too big a task to go all-at-once.

Remarks for now:
- Could you please delete unused old files, so that git's diff recognizes the file move and any subsequent changes, i.e.: `./imgs` and `./i18n`
- Could you please separate the ticker from this pull request and if you really think this should be done, PR it separately, later? This is a controversial change and we should not want discussion on that block this.
- What is the reason for `css/styles.css` to include bootstrap? I normally advice against that because the beauty of the c (cascading) part in css is that we don't need to couple - so the default of having stylesheets cascade means ease of maintenance.
- Related to the point above: Is it possible to split and mark untouched "vendor" assets from bespoke assets?
- is there a reason why some things (`./imgs`, `./i18n`) moved to `./assets` while others (like `./js` and `css`) did not?

edswf1 commented:
> Thank you! To keep momentum going, I'm going to try to give you feedback in iterations. This is going to make me look like a terrible nitpicker that is never satisfied 😂 - but the review is too big a task to go all-at-once.
> 
> Remarks for now:
> 
> * Could you please delete unused old files, so that git's diff recognizes the file move and any subsequent changes, i.e.: `./imgs` and `./i18n`
> * Could you please separate the ticker from this pull request and if you really think this should be done, PR it separately, later? This is a controversial change and we should not want discussion on that block this.
> * What is the reason for `css/styles.css` to include bootstrap? I normally advice against that because the beauty of the c (cascading) part in css is that we don't need to couple - so the default of having stylesheets cascade means ease of maintenance.
> * Related to the point above: Is it possible to split and mark untouched "vendor" assets from bespoke assets?
> * is there a reason why some things (`./imgs`, `./i18n`) moved to `./assets` while others (like `./js` and `css`) did not?

No Problem:
- deleted unused old files but kept getting started folder for .md files
- removed the ticker. 1 Dogecoin = 1 Dogecoin.
- seperated bootstrap.css and styles.css
- moved all vendor assets to assets/vendor/ . all contents of this folder are untouched
- there was no reason for keeping js and css outside of assets folder. fixed that

patricklodder commented on assets/i18n/en.json:
```diff
@@ -247,5 +247,15 @@
+    "dgc-get-some": "Get Some Dogecoin",
+    "dgc-language": "Language",
+    "dgc-back-to-main": "Back to Main Page"
+}
```
please restore the newline

edswf1 replied:
unfortunately i don't know where to restore a newline here.
Edit: sry, got it. added newline on end of file.

patricklodder commented on assets/css/styles.css:
```diff
@@ -0,0 +1,228 @@
+  .px-lg-3 {
+    padding-left: .5rem !important;
+  }
+}
```
please add a newline

edswf1 replied:
unfortunately i don't know where to add newline here.
Edit: sry got it! added new line on end of file

patricklodder commented on assets/js/scripts.js:
```diff
@@ -0,0 +1,255 @@
+    
+  })(jQuery); // End of use strict
+
+  function updatePrice(currency){
+    $.get( "https://chain.so/api/v2/get_price/DOGE/"+currency, function( response ) {
+     console.log(response);
+     var commas = 6;
+     if(currency == 'BTC'){
+      commas = 8;
+    }
+    var price = parseFloat(response.data.prices[0].price);
+    var priceBase = response.data.prices[0].price_base;
+    $selector = $(".curPrice_"+currency);
+    $selector.html(price.toFixed(commas) + " " + priceBase);
+  });
+  }
```
this should be removed

edswf1 replied:
forgot that. removed it now

patricklodder commented on guide.html:
```diff
@@ -0,0 +1,425 @@
+            <div class="collapse navbar-collapse" id="navbarResponsive">
+                <ul class="navbar-nav ml-auto">
+                    <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded" href="index.html" data-i18n="dgc-back-to-main">Back to Main Page</a></li>
+                    <li class="nav-item mx-0 mx-lg-1"><span class="nav-link py-3 px-0 px-lg-3 rounded text-white curPrice_USD"></span></li>
```
this should be removed

edswf1 replied:
i think you are referring to the curPrice_USD element. removed that now. "Back to Main Page" is part of the menu and shouldn't be removed

patricklodder commented on guide.html:
```diff
@@ -0,0 +1,425 @@
+    <script src="assets/vendor/js/jquery.i18n/jquery.i18n.language.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/jquery.i18n.parser.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/jquery.i18n.emitter.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/bs.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/dsb.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/fi.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/ga.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/he.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/hsb.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/hu.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/hy.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/la.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/ml.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/os.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/ru.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/sl.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/uk.js"></script>
```
do all of these need to be loaded even if we don't have translations? If yes, can we minify this into a single script?

edswf1 replied:
no these files where never used (imho only needed for special characters), just included them for the sake of completeness (as they are included in the current live page). removed them now form index.html and guide.html

patricklodder commented on index.html:
```diff
@@ -1,556 +1,290 @@
+                    <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#about"><span data-i18n="dgc-what-is-doge">What is dogecoin</span></a></li>
+                    <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#start"><span data-i18n="dgc-get-started-now">How to get started</span></a></li>
+                    <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded" href="guide.html"><span data-i18n="dgc-guide">Guide</span></a></li>
+                    <li class="nav-item mx-0 mx-lg-1"><a href="" class="nav-link py-3 px-0 px-lg-3 rounded text-white curPrice_USD"></a></li>
```
this needs to be removed

edswf1 replied:
i think you are referring to the curPrice_USD element. removed that now. all other elements are part of the menu and shouldn't be removed

patricklodder reviewed (commented):
went through all the files, have some minor comments.

edswf1 commented:
> went through all the files, have some minor comments.

hope i resolved the issues, menu items are now free of pricing.
sorry for the newline stuff, never heard of that convention before.

lugus commented:
> @patricklodder
> added internationalization with jQuery.i18n.
> Language is guessed by browser language (english fallback) and can be set in footer.
> Once Language is set, current Language ist stored in sessionStorage (for maintaining language when browsing to guide.html)
> added some texts to en,de,fr and es.json, however im not fluent in fr and es.
> 
> site is live at: https://edswf1.github.io/dogecoin.com-1/

French LGTM! 💯 Thanks for your work @edswf1 @patricklodder 🚀

michilumin reviewed (commented):
So I've been looking this over and, first of all, it's great, I really want to see it up soon!  The following are just ideas/suggestions though;  doing it in review just to have eyes on things:

-  /r/dogecoin isn't the only community center: "community" should link not only to reddit /r/dogecoin, but also to discord ( https://discord.gg/dogecoin ) and /r/dogeducation.
-  The only stylistic wonderings I have is the 'star' motif, since especially when it's blue, it looks very ... "american" - to a lot of people at least.  This isn't a big thing, I'm just wondering if we could replace the star with something else.  (moon?)
-  Pedantry time!  OSX isn't OSX anymore, it's now officially MacOS, so we should probably update wording to that.

... Finally, I know for a while we had a source for a dev ticker of sorts. In fact I think Langer or Ross had it working, showing the last few commits from github. I really do want to revisit this but I also really really DONT want to delay this thing going up. Because it is VERY VERY good and I don't want to get in the way - I just want to touch on that point and see how possible such a 'github repo ticker' as an element could be.

edswf1 commented:
> So I've been looking this over and, first of all, it's great, I really want to see it up soon! The following are just ideas/suggestions though; doing it in review just to have eyes on things:
> 
> * /r/dogecoin isn't the only community center: "community" should link not only to reddit /r/dogecoin, but also to discord ( https://discord.gg/dogecoin ) and /r/dogeducation.
> * The only stylistic wonderings I have is the 'star' motif, since especially when it's blue, it looks very ... "american" - to a lot of people at least.  This isn't a big thing, I'm just wondering if we could replace the star with something else.  (moon?)
> * Pedantry time!  OSX isn't OSX anymore, it's now officially MacOS, so we should probably update wording to that.
> 
> ... Finally, I know for a while we had a source for a dev ticker of sorts. In fact I think Langer or Ross had it working, showing the last few commits from github. I really do want to revisit this but I also really really DONT want to delay this thing going up. Because it is VERY VERY good and I don't want to get in the way - I just want to touch on that point and see how possible such a 'github repo ticker' as an element could be.

Thank you for your feedback. 
- added discord community at about section and footer
- changed star to angle-down (basically any font-awesome icon or image could be there)
- changed OS X to macOS

regarding github repo ticker: implementing this would be an easy task (simply parsing https://api.github.com/repos/dogecoin/dogecoin/commits response), however i would prefer doing this in a succeeding PR

patricklodder commented:
> regarding github repo ticker: implementing this would be an easy task (simply parsing https://api.github.com/repos/dogecoin/dogecoin/commits response), however i would prefer doing this in a succeeding PR

This is my preferred strategy too.

---

Two last comments before I do a final round of code review:

1. The Dogecoin coin logo changed into a Doge image, without the coin... why?
2. The creator credits at the bottom are gone. Intentional?

edswf1 commented:
> > regarding github repo ticker: implementing this would be an easy task (simply parsing https://api.github.com/repos/dogecoin/dogecoin/commits response), however i would prefer doing this in a succeeding PR
> 
> This is my preferred strategy too.
> 
> Two last comments before I do a final round of code review:
> 
> 1. The Dogecoin coin logo changed into a Doge image, without the coin... why?
> 2. The creator credits at the bottom are gone. Intentional?

1. i basically took the image of the current "what is doge" section as hero image because the coin itself is flying around in the background, is part of the menu and the favicon. Would be no problem to change that though.
2. added footer-credits again. please let me know if there should be other mentions too

patricklodder commented:
> 1. i basically took the image of the current "what is doge" section as hero image because the coin itself is flying around in the background, is part of the menu and the favicon. Would be no problem to change that though.

No this is fine, just wanted to know why.

> 2. added footer-credits again. please let me know if there should be other mentions too

Thanks! No further mentions required I think. I'll go over the 6 files that have been changed once more today.

lugus commented:
Looks like http://www.ummjackson.com is not valid anymore and should be replaced by https://ummjackson.com/

edswf1 commented:
> Looks like http://www.ummjackson.com is not valid anymore and should be replaced by https://ummjackson.com/

thanks, fixed that

nickatnight commented:
Suggestion:

Should there be a link somewhere on the page that points back to this repo?

patricklodder commented:
> Should there be a link somewhere on the page that points back to this repo?

I thought about that and I will propose a separate PR with that change after we merge this.

patricklodder reviewed (dismissed):
Full ACK. Reviewed all content & scripts. Checked all links. All things I don't like are inherited from the current site, so we will handle those outside of this PR, as they have nothing to do with the updated design.

I'm leaving this open for 12 hours before deployment (3/30, 8am PDT)

nickatnight commented:
@patricklodder just in time for the moon boost 💎 🤲

nickatnight commented:
Swell job @edswf1

lugus commented on assets/i18n/en.json:
```diff
@@ -53,7 +53,7 @@
     "dgc-stores-title": "Stores.", 
     "dgc-stores-buy": "Buy with Dogecoin.",
-    "dgc-footer-credits": "Created by <a href=\"http://www.ummjackson.com\" target=\"_blank\">Jackson Palmer</a> & <a href=\"https://twitter.com/BillyM2k\" target=\"_blank\">Shibetoshi Nakamoto</a>. Website Design by <a href=\"https://www.reddit.com/user/PowerLemons\" target=\"_blank\">/u/PowerLemons</a>. Logo by <a href=\"http://christinemix.com\" target=\"_blank\">Christine Ricks</a>.",
+    "dgc-footer-credits": "Created by <a href=\"http://www.ummjackson.com\" target=\"_blank\">Jackson Palmer</a> & <a href=\"https://twitter.com/BillyM2k\" target=\"_blank\">Shibetoshi Nakamoto</a>. Website Design by <a href=\"https://www.reddit.com/user/edswf1\" target=\"_blank\">/u/edswf1</a>. Logo by <a href=\"http://christinemix.com\" target=\"_blank\">Christine Ricks</a>.",
```
old url `http://www.ummjackson.com` here and in other translation files

patricklodder replied:
I saw those, those are also on the current site. There's also a couple more http links in translation files that need to be switched to https. I'll prepare a PR before launch so we can keep this going.

edswf1 replied:
> I saw those, those are also on the current site. There's also a couple more http links in translation files that need to be switched to https. I'll prepare a PR before launch so we can keep this going.

sorry missed your comment and updated already

lugus commented on assets/js/scripts.js:
```diff
@@ -0,0 +1,241 @@
+   } );
+}
+// Enable debug
+$.i18n.debug = true;
```
I think we should disable the debug mode?

patricklodder replied:
good catch... I missed that one, thanks! I'll PR that separately.

edswf1 replied:
> good catch... I missed that one, thanks! I'll PR that separately.

sorry missed your comment and updated already

patricklodder reviewed (approved):
Checked delta for commits ea6fb289fca323e3089a2e09be94ea26991c84b3 and 39318a4859e45321574c1204475ce1f3551fde61 since my last review. OK.

lugus commented:
Kudos to @edswf1 👏
